### ISSUE TITLE
Replace `join` with intersection, subtract etc

### ISSUE THREAD

**Issue by dmehra** [juttle/juttle#75]
We've had enough internal votes for simplifying user experience with juttle `join` to call for this ticket, and affirmed that intuition when comparing juttle primitives with Spark operations on sets. 

These (from Spark) are clear enough:
- set.union(other-set)
- set.intersection(other-set)
- set.subtract(other-set)
- set.cartesian(other-set)

These (in Juttle) are not so clear:
- `(flowgraph1; flowgraph2) | ... ` which creates an implicit union in juttle (newcomers usually try to `join` the two streams which is the wrong thing to do - perhaps we should listen to new user's intuition)
- `join` itself only does intersection, however that's changed if you use
- `join -outer N` (where N is the stream number) is a left outer join (we don't appear to support a full outer join)

Additionally, the time-related options to `join` are confusing, and are subject of another ticket.

Note that we currently do not implement 'subtract' and 'cartesian' operations. Those haven't been requested by users to date, can wait on them.
